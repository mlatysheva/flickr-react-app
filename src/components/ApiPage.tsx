import { Title } from './Home';
import { ApiSearch } from './ApiSearch';
import { PhotoList } from './ApiPhotoList';
import { ApiDashboard } from './ApiDashboard';

export function ApiPage() {
  return (
    <div className="main">
      <Title title="Find photos on Flickr" />
      <ApiSearch placeholder="Enter photo tags" />
      <ApiDashboard />
      <PhotoList />
    </div>
  );
}
