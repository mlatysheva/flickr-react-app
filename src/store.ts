import { configureStore } from '@reduxjs/toolkit';
import apiSearchReducer from './features/apiSearchSlice';
import counterReducer from './features/counterSlice';
import photosReducer from './features/photoSlice';

export const store = configureStore({
  reducer: {
    counter: counterReducer,
    apiSearch: apiSearchReducer,
    photos: photosReducer,
  },
});

export const counter = (state: { counter: { value: number } }) => state.counter.value;

export type RootState = ReturnType<typeof store.getState>;

export type AppDispatch = typeof store.dispatch;
