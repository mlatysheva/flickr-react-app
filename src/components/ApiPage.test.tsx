import { render, screen } from '@testing-library/react';
import { ApiPage } from './ApiPage';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { store } from '../store';

describe("ApiPage", () => {
  it("renders ApiPage component", () => {
    render(<Provider store={store}>
      <MemoryRouter>
      <ApiPage />
    </MemoryRouter>
    </Provider>);
    expect(screen.getByText(/Find photos on Flickr/i)).toBeInTheDocument();
    expect(screen.getByRole('button')).toBeInTheDocument();
    expect(screen.getByPlaceholderText(/Enter keywords to search for photos/i)).toBeInTheDocument();
    expect(screen.queryByText(/Enter Continent/i)).toBeNull();
    expect(screen.queryByText(/More.../i)).toBeNull();
  })
})

test ("ApiPage as fragment matches the snapshot", () => {
  const { asFragment } = render(<Provider store={store}>
    <MemoryRouter>
    <ApiPage />
  </MemoryRouter>
  </Provider>);
  expect(asFragment()).toMatchSnapshot();
})
