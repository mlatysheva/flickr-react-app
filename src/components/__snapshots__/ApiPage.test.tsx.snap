// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApiPage as fragment matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="main"
  >
    <h1>
      Find photos on Flickr
    </h1>
    <form
      class="search"
    >
      <input
        class="search-field"
        id="search"
        name="search"
        placeholder="Enter keywords to search for photos"
        type="text"
        value=""
      />
      <button
        class="search-button"
        type="submit"
      >
        Search
      </button>
    </form>
    <div
      class="lds-spinner"
    >
      <div />
      <div />
      <div />
      <div />
      <div />
      <div />
      <div />
      <div />
      <div />
      <div />
      <div />
      <div />
    </div>
  </div>
</DocumentFragment>
`;
